tf2_ros/src/static_transform_broadcaster_node.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2019, Open Source Robotics Foundation, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <memory>
+#include <stdexcept>
+#include <string>
+
+#include "tf2_ros/static_transform_broadcaster_node.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace tf2_ros
+{
+StaticTransformBroadcasterNode::StaticTransformBroadcasterNode(const rclcpp::NodeOptions & options)
+: rclcpp::Node("static_transform_publisher", options)
+// TODO(clalancette): Anonymize the node name like it is in ROS1.
+{
+  geometry_msgs::msg::TransformStamped tf_msg;
+  rcl_interfaces::msg::ParameterDescriptor descriptor;
+  descriptor.read_only = true;
+
+  tf_msg.header.stamp = this->now();
+  tf_msg.transform.translation.x = this->declare_parameter("/translation/x", 0.0, descriptor);
+  tf_msg.transform.translation.y = this->declare_parameter("/translation/y", 0.0, descriptor);
+  tf_msg.transform.translation.z = this->declare_parameter("/translation/z", 0.0, descriptor);
+  tf_msg.transform.rotation.x = this->declare_parameter("/rotation/x", 0.0, descriptor);
+  tf_msg.transform.rotation.y = this->declare_parameter("/rotation/y", 0.0, descriptor);
+  tf_msg.transform.rotation.z = this->declare_parameter("/rotation/z", 0.0, descriptor);
+  tf_msg.transform.rotation.w = this->declare_parameter("/rotation/w", 0.0, descriptor);
```

Comment:
I feel like I should also adjust this to be `1.0`, since the default quaternion is not valid here.

Reply:
Yeah, that's a great idea.